### DISCUSSION TITLE
could install Theia by a common user beside "root" user in Linux

### DISCUSSION THREAD

**Discussion by bluejade008** [eclipse-theia/theia#10452] [Ideas]
Hi all,

Could you try to install Theia via common user?
I want to limit the user access permission to Eclipse Theia, and define the special workspace?

I failed to build and complie Theia by a common user.

**Comment by msujew** [comment]
Hi @bluejade008,

I assume the building/compiling failure stems from some native dependencies which we are using. Those probably require some sort of elevated user rights.

You should be able to start Theia (both electron and browser version) as a common user after compiling it. However, as far as I know, there's a limitation regarding the access permissions: The user not only needs access to the workspace but also to the compiled backend folder of Theia, since it needs access to the compiled JavaScript sources in order to execute the backend.

**Reply by bluejade008** [reply]
I try to install nvm and nodejs via the "root" user. Subsequently create a common user, and  set up the environment of OS, and then the nvm command and nodebjs command can be used in the common user. 
I copy the theia repository form the /root/theia to the common user home directory, and keep up the package.json unchangeable, do the command " yarn && yarn download:plugins, waiting for some minutes, the building can be successful.
However if I don't do  "yarn browser build", I fail to do the command "yarn start -h xx.xx.xx.xx -p x   ，I wonder if the browser must be built firstly, and then the web IDE can be used?

**Reply by msujew** [reply]
> yarn && yarn download:plugins

Running only `yarn` will not build the application (although it does compile the TypeScript code). Only `yarn browser build` will actually run the webpack compiler, which actually builds the web-frontend. So you definitely need to run that before you can start the app.

**Reply by bluejade008** [reply]
Thanks.

One more question.

If the brower building was done, after I  do the command "yarn browser start -h xx.xx.xx.xx -p xx" in order to launch a web server,  I try to open web IDE via one kind of browser.   One menu item named ”sample menu" is also added in the IDE web title, I want to prohit the menu item. Would you like to explicate how can I prohibit the "sample menu" option? 
Thanks.

**Reply by msujew** [reply]
It looks like you are using the "Theia Example App" (which is our example app to test regressions in Theia). For a quick fix, you can simply remove this like from the package.json of the browser app:

https://github.com/eclipse-theia/theia/blob/99a56a31bc481f7382b6e6260844cbd8650a4aef/examples/browser/package.json#L22

If you want to use Theia on a more customized level, you should consider building your own Theia setup. See [the documentation for more info](https://theia-ide.org/docs/composing_applications/).

**Reply by bluejade008** [reply]
Thanks.

I have already look through Theia docu many times, however it's a pity that I still bewilder how to modify Theia extension plugins, and how to set up one kind of compute language compiler. Especially I am eager to set up C compiler in the Theia in order to compile C source code via Theia web IDE  directly. Could you give me some advice?
